### DISCUSSION TITLE
Bepu Docs

### DISCUSSION THREAD

**Discussion by VaclavElias** [stride3d/stride#2243] [Contributors]
While the integration is [still in progress](https://github.com/stride3d/stride/pull/2131), let's start a conversation on this topic. We aim to outline the structure of Bepu Docs and strive to make these new docs exemplary and comprehensive. 📘

## Key Focus Areas
There are several major parts we should focus on:
1) **XML Comments for Stride API**
2) **Stride Manual - Bepu Physics**
3) **Stride Tutorials**
4) **Blog Posts with examples**

### 1. XML Comments
Our API docs, generated by DocFX, allow adding examples and are sourced from XML comments in the code, like those shown [here](https://dotnet.github.io/docfx/docs/basic-concepts.html). We can add comprehensive remarks and examples that aren't part of the source code, avoiding clutter in the codebase. Example files are available [here](https://github.com/stride3d/stride-docs/tree/master/en/examples). These external additions are automatically merged to relevant API docs based on `uid`.

Every public member should have XML comments, and we should also utilize tools like ChatGPT/Copilot for assistance. While some content generated might be straightforward, some will need review by developers familiar with Bepu integration. 

### 2. Bepu Physics
The Bepu docs should at least mirror our [Bullet Physics docs](https://doc.stride3d.net/latest/en/manual/physics/index.html) and merge existing docs from [Nicogo1705/Stride.BepuPhysics/wiki](https://github.com/Nicogo1705/Stride.BepuPhysics/wiki). Additionally, we should utilise [Bepu docs](https://github.com/bepu/bepuphysics2).

Key topics include:

- [ ] Bepu physics
- [ ] Colliders
  - [ ] Static colliders
  - [ ] Rigidbodies
  - [ ] Kinematic rigidbodies
  - [ ] Characters
  - [ ] Collider shapes
  - [ ] Triggers
  - [ ] Constraints
- [ ] Raycasting
- [ ] Simulation 
- [ ] Tutorials
  - [ ] Creating a bouncing ball
  - [ ] Scripting a trigger

### 3. Stride Tutorials
It would be beneficial to have tutorials focused on Bepu Physics. 🚀

### 4. Blog Posts
We can create blog posts summarizing the Bepu integration and showcasing examples at a later stage. These posts can help illustrate the practical application and benefits of Bepu Physics in Stride. 📝

Let's discuss and outline our plans. Feel free to contribute ideas and suggestions! 💬

**Comment by Doprez** [comment]
Just as an added bit of info, there is no bounce or restitution out of the box.

There is a work around https://github.com/bepu/bepuphysics2/issues/52 and an example from Norbo.

**Comment by Doprez** [comment]
Lots of info is also already in the seperate Bepu repo that we can update to include in the basics of Bepu when its in Stride https://github.com/Nicogo1705/Stride.BepuPhysics/wiki

**Reply by VaclavElias** [reply]
This is actually already mentioned above, including the link. I have updated the text, so it is clearer that it point to his wiki.
